### PR TITLE
#11587 - Prevent the 'Undo deletion' of an event participant whose pe…

### LINKED ISSUE #11587: Prevent the 'Undo deletion' of an event participant whose person is listed as active event participant within the same event
### Problem Description

Currently, due to undo deletion, it is possible to have the same person listed multiple times as active event participant within the same event.

### Proposed Change

Prevent the undo deletion/Restore of an event participant if the same person is listed as an active/archived event participant within the same event.

This is needed as a person should not be allowed to have more than 1 entry as active or archived event participant within the same event.

### Acceptance Criteria

- [ ] Add a check, both in the frontend and backend so that the restore of an event participant is prohibited when the same person is already listed as active or archived event participant within that same event;
- [ ] when the user tries to restore (from the entity page or the bulk actions) such an event participant - display a warning message 'The person associated with the event participant is already listed as an active or archived event participant within the event'.

### Additional Information

Steps to reproduce the problem:
1. For Event 1, create Event participant 1 (make sure the person already has another entity associated such as a case or contact);
2. Soft delete Event participant 1;
3. Open the 'Create event participant' form and search for the person associated with Event participant 1;
4. Create the new event participant;
5. From the 'Deleted event participants' grid, undo the deletion of the event participant deleted at step 2.
Actual behavior:
The same person is now listed 2 times as event participant within the same event.

### PR DESCRIPTION
**Pull request by sergiupacurariu**
…rson is listed as active event participant within the same event

Fixes #11587

### CODE REVIEW

**Comment by sormas-vitagroup** [comment]
SonarCloud analysis: https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=13041

File: sormas-backend/src/main/java/de/symeda/sormas/backend/event/EventParticipantFacadeEjb.java [hunk 1]
```diff
@@ -564,8 +564,20 @@ public List<ProcessedEntity> restore(List<String> uuids) {
 		if (eventParticipantsToBeRestored != null) {
 			eventParticipantsToBeRestored.forEach(eventParticipantToBeRestored -> {
 				try {
-					restore(eventParticipantToBeRestored.getUuid());
-					processedEventParticipants.add(new ProcessedEntity(eventParticipantToBeRestored.getUuid(), ProcessedEntityStatus.SUCCESS));
+					final boolean empty = getByEventAndPersons(
```
Comment (leventegal-she) [inline sormas-backend/src/main/java/de/symeda/sormas/backend/event/EventParticipantFacadeEjb.java:567]: please rename to `isPersonNatParticipant` or similar

File: sormas-backend/src/main/java/de/symeda/sormas/backend/event/EventParticipantFacadeEjb.java [hunk 2]
```diff
@@ -564,8 +564,20 @@ public List<ProcessedEntity> restore(List<String> uuids) {
...
+						eventParticipantToBeRestored.getEvent().getUuid(),
+						Collections.singletonList(eventParticipantToBeRestored.getPerson().getUuid())).isEmpty();
+					if (empty) {
+						restore(eventParticipantToBeRestored.getUuid());
+						processedEventParticipants.add(new ProcessedEntity(eventParticipantToBeRestored.getUuid(), ProcessedEntityStatus.SUCCESS));
+					} else {
+						processedEventParticipants
```
[the remark this answers is not in the document]
Comment (sergiupacurariu) [reply sormas-backend/src/main/java/de/symeda/sormas/backend/event/EventParticipantFacadeEjb.java:574]: changed

File: sormas-backend/src/main/java/de/symeda/sormas/backend/event/EventParticipantFacadeEjb.java (hunk 1 above)
Comment (sergiupacurariu) [reply sormas-backend/src/main/java/de/symeda/sormas/backend/event/EventParticipantFacadeEjb.java:567]: renamed

**Comment by sormas-vitagroup** [comment]
SonarCloud analysis: https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=13041

**Review by leventegal-she — APPROVED** [review]

**Pull request merged**